File: Brad/Number_of_Islands.py
class Solution(object):
    def numIslands(self, grid):
        """
        :type grid: List[List[str]]
        :rtype: int
        """
        m = len(grid)
        n = len(grid[0])

        def visitIsland(i,j,m,n):
            grid[i][j]=0
            if i-1>=0 and grid[i-1][j]=="1":
                visitIsland(i-1,j,m,n)
            if i+1<=m-1 and grid[i+1][j]=="1":
                visitIsland(i+1,j,m,n)
            if j-1>=0 and grid[i][j-1]=="1":
                visitIsland(i,j-1,m,n)
            if j+1<=n-1 and grid[i][j+1]=="1":
                visitIsland(i,j+1,m,n)
            
        count=0
        for i in range(m):
            for j in range(n):
                if grid[i][j]=="1":
                    count+=1
                    visitIsland(i,j,m,n)
        return count

File: Kavya/Number_of_Islands.java
class Number_of_Islands {
    public int numIslands(char[][] grid) {
        int count = 0;
        for(int i = 0; i < grid.length; i++){
            for(int j = 0; j < grid[0].length; j++){
                if(grid[i][j] == '1'){
                    count++;
                    traverse(grid,i,j);
                }
            }
        }

        return count;
        
    }
    public void traverse(char [][] grid,int i ,int j){
        if(i < (grid.length - 1) && grid[i+1][j] == '1'){
             grid[i+1][j] = '0';
            traverse(grid,i+1,j);
        }

        if(j < (grid[0].length - 1) && grid[i][j+1] == '1'){
             grid[i][j+1] = '0';
            traverse(grid,i,j+1);
        }

        if(i >= 1 && grid[i-1][j] == '1'){
             grid[i-1][j] = '0';
            traverse(grid,i-1,j);
        }

        if(j >= 1 && grid[i][j-1] == '1'){
             grid[i][j-1] = '0';
            traverse(grid,i,j-1);
        }

        
        
        return;
    }
}
